dftimewolf/lib/processors/openrelik.py

```diff
@@ -0,0 +1,174 @@
+"""Processes artifacts using OpenRelik."""
+
+import tempfile
+import time
+import os
+
+from typing import Type, Iterator
+
+from openrelik_api_client import api_client, folders, workflows
+
+from dftimewolf.lib import module
+from dftimewolf.lib import state as state_lib
+from dftimewolf.lib.containers import containers, interface
+from dftimewolf.lib.modules import manager as modules_manager
+
+
+# pylint: disable=no-member
+class OpenRelikProcessor(module.ThreadAwareModule):
+  """Processes artifacts with OpenRelik."""
+
+  def __init__(
+    self,
+    state: state_lib.DFTimewolfState,
+    name: str | None = None,
+    critical: bool = False,
+  ) -> None:
+    """Initializes an OpenRelik processor.
+
+    Args:
+      state (DFTimewolfState): recipe state.
+      name (Optional[str]): The module's runtime name.
+      critical (Optional[bool]): True if the module is critical, which causes
+          the entire recipe to fail if the module encounters an error.
+    """
+    super().__init__(state=state, name=name, critical=critical)
+    self.openrelik_api_client: api_client.APIClient = None
+    self.openrelik_folder_client: folders.FoldersAPI = None
+    self.openrelik_workflow_client: workflows.WorkflowsAPI = None
+    self.openrelik_api = None
+    self.openrelik_ui = None
+    self.openrelik_api_key = None
+    self.workflow_id = None
+    self.folder_id = None
+    self.sketch_id = None
+    self.incident_id = None
+
+  # pylint: disable=arguments-differ
+  def SetUp(
+    self,
+    incident_id: str | None,
+    sketch_id: int | None,
+    folder_id: int | None,
+    workflow_id: int | None,
+    openrelik_api: str | None,
+    openrelik_ui: str | None,
+    openrelik_api_key: str | None,
+  ) -> None:
+    self.openrelik_api = openrelik_api
+    self.openrelik_ui = openrelik_ui
+    self.openrelik_api_key = openrelik_api_key
+    self.openrelik_api_client = api_client.APIClient(
+      self.openrelik_api, self.openrelik_api_key
+    )
+    self.openrelik_folder_client = folders.FoldersAPI(self.openrelik_api_client)
+    self.openrelik_workflow_client = workflows.WorkflowsAPI(
+      self.openrelik_api_client
+    )
+
+    self.folder_id = folder_id
+    self.workflow_id = workflow_id
+    self.incident_id = incident_id
+    self.sketch_id = sketch_id
+
+  def PollWorkflowStatus(self, workflow_id: int) -> Iterator[str]:
+    """Polls the status of a workflow until it completes."""
+    filename = str(workflow_id)
+    workflow = self.openrelik_workflow_client.get_workflow(
+      self.folder_id, workflow_id
+    )
+    status = workflow.get("tasks")[0].get("status_short")
+    output_file_ids = []
+    while status != "SUCCESS" and status != "FAILED":
+      self.logger.info(f"Workflow {workflow_id} status: {status}")
```

Comment:
Let's be a bit more conservative in what we're logging and move some of this to debug (especially since it can pop up many times)